src/components/dev-hub/card-list.js

```diff
@@ -65,29 +65,52 @@ const renderPodcast = podcast => (
 );
 
 const renderContentTypeCard = item => {
-    if (item.mediaType)
+    if (item.mediaType) {
         switch (item.mediaType) {
             case 'youtube':
             case 'twitch':
                 return renderVideo(item);
             case 'podcast':
                 return renderPodcast(item);
             default:
-                return;
+                return renderArticle(item);
         }
+    }
+    // Some items don't have mediaType. In that case, they are articles
     return renderArticle(item);
 };
 
 export default React.memo(
-    ({ all, videos, articles, podcasts, limit = CARD_LIST_LIMIT }) => {
+    ({
+        all,
+        videos,
+        articles,
+        podcasts,
+        textFilterQuery,
+        limit = CARD_LIST_LIMIT,
+    }) => {
         videos = videos || [];
         articles = articles || [];
         podcasts = podcasts || [];
 
         // If we provide "all", we don't need to sort. We can assume any sorting
         // has been done
-        const fullContentList =
-            all || sortCardsByDate(videos.concat(articles, podcasts));
+        let fullContentList = [];
+        if (!textFilterQuery) {
+            fullContentList = sortCardsByDate(
+                videos.concat(articles, podcasts)
+            );
```

Comment:
Do we need the entire query passed to this component or is a Boolean like `shouldSort` appropriate?

Reply:
No, we do not need the query passed, and `shouldSort` should be appropriate